4981_ChatRoom/client.cpp

```diff
@@ -264,80 +270,132 @@ void concatIP(char *ip)
     strcat(ip, tmpIP);
 }
 
-
 /********************************************************
- *  Function:       char * getUserMessage(ClientWindow *cw, char *IP, char *username)
- *                      ClientWindow *cw: ClientWindow pointer in order to access the UI
- *                      char *completeMsg: buffer that holds the complete message to send
+ *  Function:       void formatMessage(char *message, char *IP, ClientWindow *main)
+ *                      char *message: buffer that holds message to be formatted.
  *                      char *IP: string containing the IP of the client sending the message
- *                      char *username: client username
+ *                      ClientWindow *main: The UI for the client that sent the message.
  *
- *  Programmer:     Robert Arendac
+ *  Programmer:     Matt Goerwell
  *
  *  Created:        Mar 11 2017
  *
- *  Modified:
+ *  Modified:       Mar 15 2017 - strings instead of char *
+ *                  Mar 16 2017 - Including display and send method calls
  *
  *  Desc:
- *      Will get the message entered by the user into the GUI.  Will then
- *      put a string with the IP and username appended to the front into the completeMsg buffer.
+ *      This will take the message passed into it and apply standard formatting rules.
+ *      That is to say, It will prepend the message with the timestamp, username, and
+ *      IP address of the sender. It will also colour code the username eventually.
  *******************************************************/
-void getUserMessage(ClientWindow *cw, char *completeMsg, char *IP, char *username)
+void formatMessage(const char *message, const char *IP,ClientWindow *main)
 {
-    char msg[BUFLEN];
+    char timer[BUFLEN];
+    time_t timestamp;
+    struct tm * timeinfo;
 
-    cw->getUIMessage(msg);
+    time(&timestamp);
+    timeinfo = localtime(&timestamp);
+    strftime(timer,BUFLEN,"<%b %d - %R> ",timeinfo);
+    std::string msg(timer);
+    msg.append(IP);
+    msg += " - ";
+    msg.append(cltInfo.username);
+    msg += ": ";
+    msg.append(message);
```

Comment:
The alternation was due to the fact QT was complaining when I tried to just += them as one block. As for the temp string, that's probably viable, but I'd like to clarify that you mean append to msg, not message, since those are different variables.